py_zerox/zerox/models/modellitellm.py

```diff
@@ -1,50 +1,82 @@
 import os
 import aiohttp
+import warnings
+import litellm
 from typing import List, Dict, Any, Optional
 
 # Package Imports
 from .base import BaseModel
 from .types import CompletionResponse
-from ..errors import MissingOpenAIAPIKeyException
+from ..errors import ModelAccessError, NotAVisionModel, MissingEnvironmentVariables
 from ..constants.messages import Messages
+from ..constants.prompts import Prompts
 from ..processor.image import encode_image_to_base64
 
+DEFAULT_SYSTEM_PROMPT = Prompts.DEFAULT_SYSTEM_PROMPT
 
-class OpenAI(BaseModel):
-    _instances: Dict[str, "OpenAI"] = {}
-    _system_prompt = """
-    Convert the following PDF page to markdown.
-    Return only the markdown with no explanation text.
-    Do not exclude any content from the page.
-    """
+
+class litellmmodel(BaseModel):
+    ## setting the default system prompt
+    _system_prompt = DEFAULT_SYSTEM_PROMPT
 
     def __init__(
         self,
-        api_key: Optional[str] = None,
+        model: Optional[str] = None,
+        **kwargs,
     ):
-        """Initializes the OpenAI model."""
-        if api_key is None:
-            api_key = self.get_api_key()
-
-        super().__init__(api_key)
-
-    @staticmethod
-    def get_api_key(
-        api_key: Optional[str] = None,
-    ) -> str:
-        """Gets the OpenAI API key from the environment variables incase it is not provided."""
-        if api_key is None:
-            api_key = os.getenv("OPENAI_API_KEY")
-        if api_key is None:
-            raise MissingOpenAIAPIKeyException()
-        return api_key
+        """
+        Initializes the Litellm model interface.
+        :param model: The model to use for generating completions, defaults to "gpt-4o-mini". Refer: https://docs.litellm.ai/docs/providers
+        :type model: str, optional
+        
+        :param kwargs: Additional keyword arguments to pass to self.completion -> litellm.completion. Refer: https://docs.litellm.ai/docs/providers and https://docs.litellm.ai/docs/completion/input
+        """
+        super().__init__(model=model, **kwargs)
+
+        ## calling custom methods to validate the environment and model
+        self.validate_environment()
```

Comment:
> if user passes in api key through zerox object instead of environment variables this will throw an error.

@xdotli, I think you are referring the older base model class (BaseModel_bck(ABC)) in the base.py, which I no longer use in the newer codebase, I just renamed and kept it there for my reference, I'll remove it. Thanks for pointing out.